modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/skin/WebColorFieldSkin.java

```diff
@@ -81,39 +84,32 @@ public WebColorFieldSkin(final WebColorField control) {
         super.dispose();
     }
 
-    //  "^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$";
+    @Override
     protected boolean accept(String text) {
-        if (text.length() == 0) return true;
-        if (text.matches("#[a-fA-F0-9]{0,6}") || text.matches("[a-fA-F0-9]{0,6}")) {
-            return true;
-        }
-        return false;
+        return PARTIAL_PATTERN.matcher(text).matches();
     }
 
+    @Override
     protected void updateText() {
         Color color = ((WebColorField) control).getValue();
         if (color == null) color = Color.BLACK;
         getTextField().setText(Utils.formatHexString(color));
     }
 
+    @Override
     protected void updateValue() {
-        if (noChangeInValue) return;
         Color value = ((WebColorField) control).getValue();
-        String text = getTextField().getText() == null ? "" : getTextField().getText().trim().toUpperCase(Locale.ROOT);
-        if (text.matches("#[A-F0-9]{6}") || text.matches("[A-F0-9]{6}")) {
-            try {
-                Color newValue = (text.charAt(0) == '#')? Color.web(text) : Color.web("#"+text);
-                if (!newValue.equals(value)) {
-                    ((WebColorField) control).setValue(newValue);
-                } else {
-                    // calling setText results in updateValue - so we set this flag to true
-                    // so that when this is true updateValue simply returns.
-                    noChangeInValue = true;
-                    getTextField().setText(Utils.formatHexString(newValue));
-                    noChangeInValue = false;
+        String text = getTextField().getText() == null ? "" : getTextField().getText().trim();
+        if (PATTERN.matcher(text).matches()) {
+            Color newValue = (text.charAt(0) == '#') ? Color.web(text) : Color.web("#" + text);
+            if (!newValue.equals(value)) {
+                ((WebColorField) control).setValue(newValue);
+            } else {
+                String newText = Utils.formatHexString(newValue);
+
+                if (!newText.equals(text)) {
+                    getTextField().setText(newText);
                 }
-            } catch (java.lang.IllegalArgumentException ex) {
-                System.out.println("Failed to parse ["+text+"]");
```

Comment:
Removed this as it's impossible to reach this code, even in a test: `Color.web` can throw this, but since the input is fully verified by pattern, it would be a bug that should be in-your-face instead of sysout.